src/http_server.cc

```diff
@@ -1271,6 +1283,8 @@ HTTPAPIServer::InferResponseFree(
 void
 HTTPAPIServer::HandleServerHealth(evhtp_request_t* req, const std::string& kind)
 {
+  RETURN_AND_RESPOND_IF_RESTRICTED(req, "health");
```

Comment:
Can the majority of these be checked via `kind` in top level handler? ex: https://github.com/triton-inference-server/server/blob/a2762710aa3f133ae8c1abcb998b4c09c8b02194/src/http_server.cc#L4439-L4442

We already determine the "key" via regexes in top level handler, so wonder if it can just be directly passed for each matched regex or something.

```cpp
  std::string model_name, version, kind;
  if (RE2::FullMatch(
          std::string(req->uri->path->full), model_regex_, &model_name,
          &version, &kind)) {
    // NEW
    RETURN_AND_RESPOND_IF_RESTRICTED(req, kind);  // kind in [infer, stats, metadata, tracing, ...]
    // EXISTING
    if (kind == "ready") {
    ...
```

May not be a 1:1 mapping for every single one, but might cover some subsets.

Reply:
I looked at that briefly, but they don't seem to match that well - I think we'd have to create another map from kind to field. I can see moving to an enum - but I think matching in each function maybe the easiest to reason about. 

However, open to change. One thing I added in this go was static variables to only evaluate if an api is restricted once - might be harder to optimized in the top level handler ...

Reply:
There seems to be limited overlap from the regex to the names .....

https://github.com/triton-inference-server/server/blob/a2762710aa3f133ae8c1abcb998b4c09c8b02194/docs/customization_guide/inference_protocols.md#limit-endpoint-access-beta

Reply:
I'm fine with current addition per function. An enum would be nice, but just a nice-to-have. Just wanted to see if this brought up any "lightbulbs" on current organization. Will resolve this thread.